Core/DISET/private/InnerRPCClient.py

```diff
@@ -5,26 +5,26 @@
 
 from DIRAC.Core.DISET.private.BaseClient import BaseClient
 from DIRAC.Core.Utilities.ReturnValues import S_OK
-
+from DIRAC.Core.Utilities.DErrno import cmpError, ENOAUTH
 
 class InnerRPCClient( BaseClient ):
 
   __retry = 0
 
   def executeRPC( self, functionName, args ):
-    stub = ( self._getBaseStub(), functionName, args )
     retVal = self._connect()
+    stub = ( self._getBaseStub(), functionName, args )
     if not retVal[ 'OK' ]:
       retVal[ 'rpcStub' ] = stub
       return retVal
     trid, transport = retVal[ 'Value' ]
     try:
       retVal = self._proposeAction( transport, ( "RPC", functionName ) )
       if not retVal['OK']:
-        if retVal['Message'] == "Unauthorized query":  # TODO: DErno will help!:
+        if cmpError( retVal, ENOAUTH ):  # This query is unauthorized
```

Comment:
May be. However I think, that DISET authorization policies deserve their own error code. Standard errors are rather for accessing resources like disk directories, files, etc.